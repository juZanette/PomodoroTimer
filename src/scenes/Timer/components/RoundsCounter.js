import React from 'react'
import styled from 'styled-components'
import { useSelector } from 'react-redux'

const ROUNDS_NUMBER = 4

const CounterLabel = styled.span`
  font-weight: bold;
`

const RoundsCounter = () => {
  const { currentRound } = useSelector((state) => state.timer)

  return (
    <CounterLabel>
      {currentRound}/{ROUNDS_NUMBER}
    </CounterLabel>
  )
}

export { RoundsCounter }
