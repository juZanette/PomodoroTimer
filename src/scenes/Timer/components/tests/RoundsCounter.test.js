import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { createMount } from '@material-ui/core/test-utils'
import { RoundsCounter, CounterLabel } from '../RoundsCounter'

describe('<RoundsCounter />', () => {
  const mockStore = configureMockStore([thunk])

  let store
  let mount
  let wrapper

  beforeEach(() => {
    mount = createMount()

    const storeData = {
      timer: { currentRound: 3, timeLeft: { minutes: 25, seconds: 0 } },
      settings: { rounds: 4 },
    }

    store = mockStore(storeData)
    store.dispatch = jest.fn()

    wrapper = mount(
      <Provider store={store}>
        <RoundsCounter />
      </Provider>
    )
  })

  test('should render <RoundsCounter /> correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should render current rounds/total rounds', () => {
    expect(wrapper.find(CounterLabel).text()).toBe('3/4')
  })
})
