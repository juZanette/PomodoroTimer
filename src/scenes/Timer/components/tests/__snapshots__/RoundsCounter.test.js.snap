// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoundsCounter /> should render <RoundsCounter /> correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <RoundsCounter>
    <styled.span>
      <span
        className="sc-AxjAm gfJUZX"
      >
        3
        /
        4
      </span>
    </styled.span>
  </RoundsCounter>
</Provider>
`;
